newm-server/src/main/kotlin/io/newm/server/ktx/InterProcessMutexExt.kt

```diff
@@ -0,0 +1,35 @@
+package io.newm.server.ktx
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.withContext
+import org.apache.curator.framework.recipes.locks.InterProcessMutex
+
+// InterProcessMutext does not play nice with coroutines if the block() resumes on a different thread
+// This extension function ensures that the block() is executed on the same thread as the lock and
+// releases the lock after the block is executed on the same thread.
+
+@OptIn(ExperimentalCoroutinesApi::class)
```

Comment:
Should we still be using `ExperimentalCoroutinesApi` ? Curious if we need to update kotlin versions